[api] xpath_engine: support project.remoteurl.

jberry-suse opened the pull request:
The intended use is to find projects that have a `remoteurl` to avoid static config needed by https://github.com/openSUSE/osc-plugin-factory/blob/master/check_source_in_factory.py. It works if each root-level project is checked by loading the `_meta` and checking for `remoteurl`, but it would be nice to be able to search for all such projects at once.

It looked simple enough to add since it seems to be in the same place as `url`, but if this is not sufficient let me know as I did not setup a development environment.

ChrisBr commented:
LGTM :+1: 

@adrianschroeter what do you think?

adrianschroeter commented:
The commit is fine (just the missing [api] tag), but I like to see also some test case for it.

It should be easy to extend

 src/api/test/functional/search_controller_test.rb

for this.

jberry-suse commented:
Added `[api]` tag to commit message.

That reminds me, when is the preferred method to check if "not-empty" or "exists"? My thought would be `remoteurl` or `[remoteurl]` since it just needs to be found. You can see the test I added (tried starts-with for kicks after), but seems to be crashing instead of returning a result.

hennevogel commented:
@jberry-suse sounds like you want to run the test locally and look into the logs :-)

hennevogel commented:
@jberry-suse are you going to address the test suite failures? If you need help let us know :-)

jberry-suse commented:
Yeah, I have been busy with other things, but I need to get an OBS development environment up for a few things now. I attempted it, but the vargrant setup recommended did not work on Tumbleweed and even after building vagrant and such from scratch it still had other issues.

hennevogel commented:
@jberry-suse vagrant works for me on TW. Did you use the TW package or the [official package](https://releases.hashicorp.com/vagrant/1.9.1/vagrant_1.9.1_x86_64.rpm)? :-)

jberry-suse commented:
The vagrant setup works after a TW update with vagrant from git. Not sure why official package does not want to work for me. Anyway, I fixed the test so this should be good to go.

jberry-suse commented:
Seems to have an unrelated failure:
```
Capybara::ElementNotFound:         Capybara::ElementNotFound: Unable to find link "submit request"
```

hennevogel commented:
@jberry-suse I doubt this is unrelated. The tests are also about submit requests :-)

adrianschroeter commented:
yes, the error is independend.

I have simplified your submission anyway a bit more here: https://github.com/openSUSE/open-build-service/pull/2760

Welcome new OBS developer! :)
